### DISCUSSION TITLE
Nestjs | Vercel | esn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.

### DISCUSSION THREAD

**Discussion by siamahnaf** [vercel/community#2622] [Help]
### Summary

Hello, I need to deploy my nestjs application into vercel.

Here is my `vercel.json`-

```
{
    "version": 2,
    "builds": [
        {
            "src": "src/main.ts",
            "use": "@vercel/node"
        }
    ],
    "routes": [
        {
            "src": "/(.*)",
            "dest": "src/main.ts",
            "methods": [
                "GET",
                "POST",
                "PUT",
                "DELETE"
            ],
            "headers": {
                "Access-Control-Allow-Credentials": "true",
                "Access-Control-Allow-Origin": "*",
                "Access-Control-Allow-Headers": "X-CSRF-Token, X-Requested-With, Accept, Accept-Version, Content-Length, Content-MD5, Content-Type, Date, X-Api-Version"
            }
        }
    ]
}
```

And In nestjs I already enable cors-

```
async function bootstrap() {
  const app = await NestFactory.create(AppModule);
  app.useGlobalPipes(new ValidationPipe());
  app.enableCors({
    origin: "*",
    credentials: true
  });
  await app.listen(3000);
}
bootstrap();
```

But Every time I am getting cors error-

`Access to XMLHttpRequest at  'https://flimybook-server.vercel.app/flimybook' from origin 'http://localhost:3000' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.`

Then how can I enable cors when I deploy it into vercel.

Note: If I run my server into locale I am not getting any cors errors, And also I already deploy it into `railway`, cyclic.sh` and `aws`, but I am getting any cors error. Errors occured when I am try to deploy it into vercel.

### Steps to Reproduce

No Item

**Comment by Xantino1997** [comment]
when passing credentials:true, the origin must be the URL in question, it cannot be "*", I hope the comment helps you

**Reply by siamahnaf** [reply]
Not working!

**Comment by Xantino1997** [comment]
{
  "version": 2,
  "builds": [
    {
      "src": "./index.js",
      "use": "@vercel/node"
    }
  ],
  "rewrites": [
    {
      "source": "/(.*)",
      "destination": "/index.js"
    }
  ],
  "headers": [
    {
      "source": "/(.*)",
      "headers": [
        {
          "key": "Access-Control-Allow-Origin",
          "value": "https://example.com"
        },
        {
          "key": "Access-Control-Allow-Methods",
          "value": "GET,POST,PUT,DELETE,OPTIONS"
        },
        {
          "key": "Access-Control-Allow-Headers",
          "value": "X-Requested-With, Content-Type, Accept, Authorization"
        },
        {
          "key": "Access-Control-Allow-Credentials",
          "value": "true"
        }
      ]
    }
  ]
}

this worked for me , hope it helps you
